File: mobile application development/front/kotlin/app/src/main/java/com/itmo/mybroker/data/Market.kt
package com.itmo.mybroker.data

import kotlin.math.abs
import kotlin.math.max
import kotlin.math.min

object Market {

    private fun mulberry32(seedIn: Int): () -> Double {
        var a = seedIn
        return {
            a += 0x6D2B79F5
            var t = a
            t = (t xor (t ushr 15)) * (t or 1)
            t = t xor (t + (t xor (t ushr 7)) * (t or 61))

            ((t xor (t ushr 14)).toLong() and 0xFFFFFFFFL).toDouble() / 4294967296.0
        }
    }

    private fun strSeed(s: String): Int {

        var h = 0x811C9DC5L.toInt()
        for (ch in s) {
            h = (h xor ch.code) * 16777619
        }
        return h
    }

    val TICKERS: List<Ticker> = listOf(
        Ticker("SBER",  MarketId.MOEX,   "Сбербанк",     "Sberbank",            "Банки",         "Banks",       312.45,  0.018, 0.0006,  Ccy.RUB),
        Ticker("GAZP",  MarketId.MOEX,   "Газпром",      "Gazprom",             "Энергетика",    "Energy",      138.62,  0.022, -0.0002, Ccy.RUB),
        Ticker("LKOH",  MarketId.MOEX,   "Лукойл",       "Lukoil",              "Энергетика",    "Energy",      7642.5,  0.014, 0.0003,  Ccy.RUB),
        Ticker("YDEX",  MarketId.MOEX,   "Яндекс",       "Yandex",              "Технологии",    "Tech",        4378.0,  0.026, 0.0010,  Ccy.RUB),
        Ticker("T",     MarketId.MOEX,   "Т-Технологии", "T-Technologies",      "Финансы",       "Finance",     3215.6,  0.024, 0.0007,  Ccy.RUB),
        Ticker("GMKN",  MarketId.MOEX,   "Норникель",    "Nornickel",           "Металлургия",   "Metals",      132.84,  0.020, -0.0001, Ccy.RUB),
        Ticker("ROSN",  MarketId.MOEX,   "Роснефть",     "Rosneft",             "Энергетика",    "Energy",      588.30,  0.017, 0.0001,  Ccy.RUB),
        Ticker("NVTK",  MarketId.MOEX,   "Новатэк",      "Novatek",             "Энергетика",    "Energy",      1234.5,  0.019, -0.0003, Ccy.RUB),
        Ticker("AAPL",  MarketId.NASDAQ, "Apple",        "Apple Inc.",          "Технологии",    "Tech",        218.74,  0.014, 0.0004,  Ccy.USD),
        Ticker("MSFT",  MarketId.NASDAQ, "Microsoft",    "Microsoft Corp.",     "Технологии",    "Tech",        462.18,  0.013, 0.0005,  Ccy.USD),
        Ticker("NVDA",  MarketId.NASDAQ, "NVIDIA",       "NVIDIA Corp.",        "Полупроводники","Semis",       138.92,  0.030, 0.0014,  Ccy.USD),
        Ticker("TSLA",  MarketId.NASDAQ, "Tesla",        "Tesla Inc.",          "Авто",          "Auto",        248.50,  0.034, -0.0003, Ccy.USD),
        Ticker("GOOGL", MarketId.NASDAQ, "Alphabet",     "Alphabet Inc.",       "Технологии",    "Tech",        178.20,  0.016, 0.0004,  Ccy.USD),
        Ticker("AMZN",  MarketId.NASDAQ, "Amazon",       "Amazon.com",          "E-commerce",    "E-commerce",  196.85,  0.018, 0.0005,  Ccy.USD),
        Ticker("META",  MarketId.NASDAQ, "Meta",         "Meta Platforms",      "Технологии",    "Tech",        528.60,  0.020, 0.0006,  Ccy.USD),
        Ticker("AMD",   MarketId.NASDAQ, "AMD",          "Advanced Micro Devices", "Полупроводники", "Semis",    156.30,  0.028, 0.0008,  Ccy.USD),
    )

    object FX {
        const val RUB_USD = 0.0103
        const val RUB_EUR = 0.0096
        const val USD_RUB = 97.05
        const val EUR_RUB = 104.8
        const val USD_EUR = 0.93
        const val EUR_USD = 1.075
    }

    private fun tfMs(tf: String): Long = when (tf) {
        "5m" -> 5L * 60 * 1000
        "30m" -> 30L * 60 * 1000
        "1h" -> 60L * 60 * 1000
        "1d" -> 24L * 60 * 60 * 1000
        "1w" -> 7L * 24 * 60 * 60 * 1000
        else -> 24L * 60 * 60 * 1000
    }

    private val candleCache = HashMap<String, List<Candle>>()

    fun genCandles(t: Ticker, tf: String = "1d", n: Int = 240): List<Candle> {
        val key = "${t.sym}:$tf:$n"
        candleCache[key]?.let { return it }

        val ms = tfMs(tf)
        val rand = mulberry32(strSeed("${t.sym}:$tf"))
        val out = ArrayList<Candle>(n)
        var price = t.base * (0.85 + rand() * 0.3)
        val volBase = (t.base * 0.001) * (if (t.sym == "GAZP") 5e6 else 2e5)

        val now = utc(2026, 5, 8, 9, 29)
        val start = now - n * ms
        for (i in 0 until n) {
            val time = start + i * ms

            var g = 0.0
            repeat(6) { g += rand() }
            g = (g - 3.0) / 1.732
            val ret = t.drift + t.vol * g
            val open = price
            val close = open * (1 + ret)
            val hRet = abs(t.vol * (rand() * 1.5 + 0.2))
            val lRet = abs(t.vol * (rand() * 1.5 + 0.2))
            val high = max(open, close) * (1 + hRet)
            val low  = min(open, close) * (1 - lRet)
            val v = volBase * (0.4 + rand() * 1.8) * (1 + abs(g) * 0.6)
            out.add(Candle(time, open, high, low, close, v))
            price = close
        }

        val last = out.last()
        val adjust = t.base / last.c
        if (abs(1 - adjust) < 0.4) {
            val k = min(6, out.size)
            for (i in (out.size - k) until out.size) {
                val w = (i - (out.size - k) + 1).toDouble() / k
                val m = 1 + (adjust - 1) * w
                val b = out[i]
                out[i] = b.copy(o = b.o * m, c = b.c * m, h = b.h * m, l = b.l * m)
            }
        }
        candleCache[key] = out
        return out
    }

    fun dayStats(t: Ticker): DayStats {
        val c = genCandles(t, "1d", 60)
        val last = c[c.size - 1]
        val prev = c[c.size - 2]
        val change = last.c - prev.c
        val changePct = change / prev.c * 100
        val spark = c.subList(c.size - 30, c.size).map { it.c }
        return DayStats(price = last.c, change = change, changePct = changePct, spark = spark, candles = c)
    }

    fun genOrderbook(t: Ticker, levels: Int = 12): Orderbook {
        val stats = dayStats(t)
        val mid = stats.price
        val tick = when {
            mid > 1000 -> 0.5
            mid > 100 -> 0.05
            else -> 0.01
        }
        val rand = mulberry32(strSeed("${t.sym}:ob"))
        val bids = ArrayList<BookLevel>(levels)
        val asks = ArrayList<BookLevel>(levels)
        for (i in 0 until levels) {
            val bp = mid - tick * (i + 1) - rand() * tick * 0.4
            val ap = mid + tick * (i + 1) + rand() * tick * 0.4
            val baseSize = (50 + rand() * 950).toInt()
            val depthFactor = 1 + i * 0.15 + rand() * 0.4
            bids.add(BookLevel(bp, (baseSize * depthFactor).toInt()))
            asks.add(BookLevel(ap, (baseSize * depthFactor * (0.7 + rand() * 0.6)).toInt()))
        }
        return Orderbook(bids = bids, asks = asks, mid = mid, tick = tick)
    }

    fun bySym(sym: String): Ticker = TICKERS.firstOrNull { it.sym == sym } ?: TICKERS.first()

    val PORTFOLIO_HOLDINGS: List<Holding> = listOf(
        Holding("SBER", 240, 285.40),
        Holding("YDEX", 8,   4012.00),
        Holding("LKOH", 3,   7320.00),
        Holding("NVDA", 14,  121.20),
        Holding("AAPL", 22,  198.40),
        Holding("T",    18,  3105.00),
    )

    const val CASH_RUB: Double = 184_320.0

    val HISTORY: List<Operation> = listOf(
        Operation("op-2024", utc(2026, 5, 8, 8, 12),  OpType.Buy,    "NVDA", 4,   138.92,  OpStatus.Filled,    Ccy.USD),
        Operation("op-2023", utc(2026, 5, 7, 16, 45), OpType.Sell,   "GAZP", 100, 138.62,  OpStatus.Filled,    Ccy.RUB),
        Operation("op-2022", utc(2026, 5, 7, 12, 30), OpType.Buy,    "YDEX", 2,   4368.50, OpStatus.Partial,   Ccy.RUB, filledQty = 1),
        Operation("op-2021", utc(2026, 5, 6, 14, 15), OpType.Buy,    "SBER", 60,  309.80,  OpStatus.Filled,    Ccy.RUB),
        Operation("op-2020", utc(2026, 5, 6, 10, 2),  OpType.Cancel, "TSLA", 5,   250.00,  OpStatus.Cancelled, Ccy.USD),
        Operation("op-2019", utc(2026, 5, 5, 18, 22), OpType.Sell,   "LKOH", 1,   7615.00, OpStatus.Filled,    Ccy.RUB),
        Operation("op-2018", utc(2026, 5, 5, 11, 50), OpType.Buy,    "AAPL", 6,   215.30,  OpStatus.Filled,    Ccy.USD),
        Operation("op-2017", utc(2026, 5, 4, 15, 10), OpType.Buy,    "T",    5,   3201.00, OpStatus.Filled,    Ccy.RUB),
        Operation("op-2016", utc(2026, 5, 3, 17, 35), OpType.Reject, "NVDA", 50,  138.92,  OpStatus.Rejected,  Ccy.USD, reason = "Недостаточно средств"),
    )

    val ACTIVE_ORDERS: List<ActiveOrder> = listOf(
        ActiveOrder("lo-501", "SBER", TradeSide.Buy,  100, 308.00, 0, utc(2026, 5, 8, 9, 0)),
        ActiveOrder("lo-502", "GMKN", TradeSide.Sell, 50,  135.00, 0, utc(2026, 5, 7, 14, 30)),
    )

    val USER = UserProfile(
        initials = "МА",
        nameRu = "Михаил Ахметов",
        nameEn = "Mikhail Akhmetov",
        email = "m.akhmetov@itmo.ru",
        memberSinceRu = "12 окт 2024",
        memberSinceEn = "Oct 12, 2024",
        ordersCount = "183",
        streak = "14d",
    )

    val INDICES: List<MarketIndex> = listOf(
        MarketIndex("MOEX", "3 248,17"),
        MarketIndex("NASDAQ", "21 184,3"),
    )

    private fun utc(year: Int, month1: Int, day: Int, hour: Int, minute: Int): Long {
        val cal = java.util.Calendar.getInstance(java.util.TimeZone.getTimeZone("UTC"))
        cal.clear()
        cal.set(year, month1 - 1, day, hour, minute, 0)
        cal.set(java.util.Calendar.MILLISECOND, 0)
        return cal.timeInMillis
    }
}

File: mobile application development/front/react-native/src/data/market.ts


export type Market = 'MOEX' | 'NASDAQ';
export type Ccy = 'RUB' | 'USD' | 'EUR';

export type Ticker = {
  sym: string;
  market: Market;
  nameRu: string;
  nameEn: string;
  sector: string;
  sectorEn: string;
  base: number;
  vol: number;
  drift: number;
  ccy: Ccy;
};

export type Candle = { t: number; o: number; h: number; l: number; c: number; v: number };

export type Tf = '5m' | '30m' | '1h' | '1d' | '1w';

function imul(a: number, b: number): number {

  return Math.imul(a | 0, b | 0);
}

export function mulberry32(seed: number): () => number {
  let a = seed >>> 0;
  return function () {
    a = (a + 0x6d2b79f5) >>> 0;
    let t = a;
    t = imul(t ^ (t >>> 15), t | 1);
    t ^= t + imul(t ^ (t >>> 7), t | 61);
    return ((t ^ (t >>> 14)) >>> 0) / 4294967296;
  };
}

export function strSeed(s: string): number {
  let h = 2166136261;
  for (let i = 0; i < s.length; i++) h = imul(h ^ s.charCodeAt(i), 16777619);
  return h >>> 0;
}

export const TICKERS: Ticker[] = [
  { sym: 'SBER',  market: 'MOEX',   nameRu: 'Сбербанк',     nameEn: 'Sberbank',           sector: 'Банки',           sectorEn: 'Banks',        base: 312.45, vol: 0.018, drift:  0.0006, ccy: 'RUB' },
  { sym: 'GAZP',  market: 'MOEX',   nameRu: 'Газпром',      nameEn: 'Gazprom',            sector: 'Энергетика',      sectorEn: 'Energy',       base: 138.62, vol: 0.022, drift: -0.0002, ccy: 'RUB' },
  { sym: 'LKOH',  market: 'MOEX',   nameRu: 'Лукойл',       nameEn: 'Lukoil',             sector: 'Энергетика',      sectorEn: 'Energy',       base: 7642.5, vol: 0.014, drift:  0.0003, ccy: 'RUB' },
  { sym: 'YDEX',  market: 'MOEX',   nameRu: 'Яндекс',       nameEn: 'Yandex',             sector: 'Технологии',      sectorEn: 'Tech',         base: 4378.0, vol: 0.026, drift:  0.0010, ccy: 'RUB' },
  { sym: 'T',     market: 'MOEX',   nameRu: 'Т-Технологии', nameEn: 'T-Technologies',     sector: 'Финансы',         sectorEn: 'Finance',      base: 3215.6, vol: 0.024, drift:  0.0007, ccy: 'RUB' },
  { sym: 'GMKN',  market: 'MOEX',   nameRu: 'Норникель',    nameEn: 'Nornickel',          sector: 'Металлургия',     sectorEn: 'Metals',       base: 132.84, vol: 0.020, drift: -0.0001, ccy: 'RUB' },
  { sym: 'ROSN',  market: 'MOEX',   nameRu: 'Роснефть',     nameEn: 'Rosneft',            sector: 'Энергетика',      sectorEn: 'Energy',       base: 588.30, vol: 0.017, drift:  0.0001, ccy: 'RUB' },
  { sym: 'NVTK',  market: 'MOEX',   nameRu: 'Новатэк',      nameEn: 'Novatek',            sector: 'Энергетика',      sectorEn: 'Energy',       base: 1234.5, vol: 0.019, drift: -0.0003, ccy: 'RUB' },
  { sym: 'AAPL',  market: 'NASDAQ', nameRu: 'Apple',        nameEn: 'Apple Inc.',         sector: 'Технологии',      sectorEn: 'Tech',         base: 218.74, vol: 0.014, drift:  0.0004, ccy: 'USD' },
  { sym: 'MSFT',  market: 'NASDAQ', nameRu: 'Microsoft',    nameEn: 'Microsoft Corp.',    sector: 'Технологии',      sectorEn: 'Tech',         base: 462.18, vol: 0.013, drift:  0.0005, ccy: 'USD' },
  { sym: 'NVDA',  market: 'NASDAQ', nameRu: 'NVIDIA',       nameEn: 'NVIDIA Corp.',       sector: 'Полупроводники',  sectorEn: 'Semis',        base: 138.92, vol: 0.030, drift:  0.0014, ccy: 'USD' },
  { sym: 'TSLA',  market: 'NASDAQ', nameRu: 'Tesla',        nameEn: 'Tesla Inc.',         sector: 'Авто',            sectorEn: 'Auto',         base: 248.50, vol: 0.034, drift: -0.0003, ccy: 'USD' },
  { sym: 'GOOGL', market: 'NASDAQ', nameRu: 'Alphabet',     nameEn: 'Alphabet Inc.',      sector: 'Технологии',      sectorEn: 'Tech',         base: 178.20, vol: 0.016, drift:  0.0004, ccy: 'USD' },
  { sym: 'AMZN',  market: 'NASDAQ', nameRu: 'Amazon',       nameEn: 'Amazon.com',         sector: 'E-commerce',      sectorEn: 'E-commerce',   base: 196.85, vol: 0.018, drift:  0.0005, ccy: 'USD' },
  { sym: 'META',  market: 'NASDAQ', nameRu: 'Meta',         nameEn: 'Meta Platforms',     sector: 'Технологии',      sectorEn: 'Tech',         base: 528.60, vol: 0.020, drift:  0.0006, ccy: 'USD' },
  { sym: 'AMD',   market: 'NASDAQ', nameRu: 'AMD',          nameEn: 'Advanced Micro Devices', sector: 'Полупроводники', sectorEn: 'Semis',     base: 156.30, vol: 0.028, drift:  0.0008, ccy: 'USD' },
];

export const FX = {
  RUB_USD: 0.0103,
  RUB_EUR: 0.0096,
  USD_RUB: 97.05,
  EUR_RUB: 104.8,
  USD_EUR: 0.93,
  EUR_USD: 1.075,
};

export function genCandles(t: Ticker, tf: Tf = '1d', n = 240): Candle[] {
  const msMap: Record<Tf, number> = {
    '5m': 5 * 60e3,
    '30m': 30 * 60e3,
    '1h': 60 * 60e3,
    '1d': 24 * 60 * 60e3,
    '1w': 7 * 24 * 60 * 60e3,
  };
  const ms = msMap[tf] || 24 * 60 * 60e3;
  const seedKey = t.sym + ':' + tf;
  const rand = mulberry32(strSeed(seedKey));
  const candles: Candle[] = [];
  let price = t.base * (0.85 + rand() * 0.3);
  const now = Date.UTC(2026, 4, 8, 9, 30) - 60e3;
  const start = now - n * ms;
  const volBase = t.base * 0.001 * (t.sym === 'GAZP' ? 5e6 : 2e5);
  for (let i = 0; i < n; i++) {
    const time = start + i * ms;
    let g = 0;
    for (let k = 0; k < 6; k++) g += rand();
    g = (g - 3) / 1.732;
    const ret = t.drift + t.vol * g;
    const open = price;
    const close = open * (1 + ret);
    const hRet = Math.abs(t.vol * (rand() * 1.5 + 0.2));
    const lRet = Math.abs(t.vol * (rand() * 1.5 + 0.2));
    const high = Math.max(open, close) * (1 + hRet);
    const low = Math.min(open, close) * (1 - lRet);
    const v = volBase * (0.4 + rand() * 1.8) * (1 + Math.abs(g) * 0.6);
    candles.push({ t: time, o: open, h: high, l: low, c: close, v });
    price = close;
  }
  const last = candles[candles.length - 1];
  const adjust = t.base / last.c;
  if (Math.abs(1 - adjust) < 0.4) {
    const k = Math.min(6, candles.length);
    for (let i = candles.length - k; i < candles.length; i++) {
      const w = (i - (candles.length - k) + 1) / k;
      const m = 1 + (adjust - 1) * w;
      candles[i].o *= m;
      candles[i].c *= m;
      candles[i].h *= m;
      candles[i].l *= m;
    }
  }
  return candles;
}

export type DayStats = {
  price: number;
  change: number;
  changePct: number;
  spark: number[];
  candles: Candle[];
};

export function dayStats(t: Ticker): DayStats {
  const c = genCandles(t, '1d', 60);
  const last = c[c.length - 1];
  const prev = c[c.length - 2];
  const change = last.c - prev.c;
  const changePct = (change / prev.c) * 100;
  const spark = c.slice(-30).map((b) => b.c);
  return { price: last.c, change, changePct, spark, candles: c };
}

export type ObLevel = { price: number; size: number };
export type Orderbook = { bids: ObLevel[]; asks: ObLevel[]; mid: number; tick: number };

export function genOrderbook(t: Ticker, levels = 12): Orderbook {
  const stats = dayStats(t);
  const mid = stats.price;
  const tick = mid > 1000 ? 0.5 : mid > 100 ? 0.05 : 0.01;
  const rand = mulberry32(strSeed(t.sym + ':ob'));
  const bids: ObLevel[] = [];
  const asks: ObLevel[] = [];
  for (let i = 0; i < levels; i++) {
    const bp = mid - tick * (i + 1) - rand() * tick * 0.4;
    const ap = mid + tick * (i + 1) + rand() * tick * 0.4;
    const baseSize = Math.round(50 + rand() * 950);
    const depthFactor = 1 + i * 0.15 + rand() * 0.4;
    bids.push({ price: bp, size: Math.round(baseSize * depthFactor) });
    asks.push({ price: ap, size: Math.round(baseSize * depthFactor * (0.7 + rand() * 0.6)) });
  }
  return { bids, asks, mid, tick };
}

export type Holding = { sym: string; qty: number; avg: number };
export const PORTFOLIO_HOLDINGS: Holding[] = [
  { sym: 'SBER', qty: 240, avg: 285.4 },
  { sym: 'YDEX', qty: 8, avg: 4012.0 },
  { sym: 'LKOH', qty: 3, avg: 7320.0 },
  { sym: 'NVDA', qty: 14, avg: 121.2 },
  { sym: 'AAPL', qty: 22, avg: 198.4 },
  { sym: 'T', qty: 18, avg: 3105.0 },
];

export const CASH_RUB = 184_320;

export type HistoryStatus = 'filled' | 'partial' | 'cancelled' | 'rejected' | 'pending';
export type HistoryType = 'buy' | 'sell' | 'cancel' | 'reject';
export type HistoryItem = {
  id: string;
  t: number;
  type: HistoryType;
  sym: string;
  qty: number;
  price: number;
  status: HistoryStatus;
  currency: Ccy;
  filledQty?: number;
  reason?: string;
};

export const HISTORY: HistoryItem[] = [
  { id: 'op-2024', t: Date.UTC(2026, 4, 8, 8, 12),  type: 'buy',    sym: 'NVDA', qty: 4,   price: 138.92,  status: 'filled',    currency: 'USD' },
  { id: 'op-2023', t: Date.UTC(2026, 4, 7, 16, 45), type: 'sell',   sym: 'GAZP', qty: 100, price: 138.62,  status: 'filled',    currency: 'RUB' },
  { id: 'op-2022', t: Date.UTC(2026, 4, 7, 12, 30), type: 'buy',    sym: 'YDEX', qty: 2,   price: 4368.50, status: 'partial',   currency: 'RUB', filledQty: 1 },
  { id: 'op-2021', t: Date.UTC(2026, 4, 6, 14, 15), type: 'buy',    sym: 'SBER', qty: 60,  price: 309.80,  status: 'filled',    currency: 'RUB' },
  { id: 'op-2020', t: Date.UTC(2026, 4, 6, 10, 2),  type: 'cancel', sym: 'TSLA', qty: 5,   price: 250.00,  status: 'cancelled', currency: 'USD' },
  { id: 'op-2019', t: Date.UTC(2026, 4, 5, 18, 22), type: 'sell',   sym: 'LKOH', qty: 1,   price: 7615.00, status: 'filled',    currency: 'RUB' },
  { id: 'op-2018', t: Date.UTC(2026, 4, 5, 11, 50), type: 'buy',    sym: 'AAPL', qty: 6,   price: 215.30,  status: 'filled',    currency: 'USD' },
  { id: 'op-2017', t: Date.UTC(2026, 4, 4, 15, 10), type: 'buy',    sym: 'T',    qty: 5,   price: 3201.00, status: 'filled',    currency: 'RUB' },
  { id: 'op-2016', t: Date.UTC(2026, 4, 3, 17, 35), type: 'reject', sym: 'NVDA', qty: 50,  price: 138.92,  status: 'rejected',  currency: 'USD', reason: 'Недостаточно средств' },
];

export type ActiveOrder = {
  id: string;
  sym: string;
  side: 'buy' | 'sell';
  qty: number;
  limit: number;
  filled: number;
  placedAt: number;
};

export const ACTIVE_ORDERS: ActiveOrder[] = [
  { id: 'lo-501', sym: 'SBER', side: 'buy',  qty: 100, limit: 308.00, filled: 0, placedAt: Date.UTC(2026, 4, 8, 9, 0) },
  { id: 'lo-502', sym: 'GMKN', side: 'sell', qty: 50,  limit: 135.00, filled: 0, placedAt: Date.UTC(2026, 4, 7, 14, 30) },
];

export function findTicker(sym: string): Ticker {
  const t = TICKERS.find((x) => x.sym === sym);
  if (!t) {
    return {
      sym,
      market: 'MOEX',
      nameRu: sym,
      nameEn: sym,
      sector: 'Неизвестно',
      sectorEn: 'Unknown',
      base: 100,
      vol: 0.02,
      drift: 0,
      ccy: 'RUB',
    };
  }
  return t;
}

export type UserProfile = {
  initials: string;
  nameRu: string;
  nameEn: string;
  email: string;
  memberSinceRu: string;
  memberSinceEn: string;
  ordersCount: string;
  streak: string;
};

export type MarketIndex = { label: string; value: string };
